examples/showcase/post-processing/bloom.js

```diff
@@ -0,0 +1,97 @@
+/* global THREE */
+/**
+ * Unreal Bloom Effect
+ * Implementation for A-Frame
+ * Code modified from Akbartus's post-processing A-Frame integration
+ * https://github.com/akbartus/A-Frame-Component-Postprocessing
+ */
+
+import AFRAME from "aframe";
+import { EffectComposer } from "three/addons/postprocessing/EffectComposer.js";
+import { RenderPass } from "three/addons/postprocessing/RenderPass.js";
+import { UnrealBloomPass } from "three/addons/postprocessing/UnrealBloomPass.js";
+import { OutputPass } from "three/addons/postprocessing/OutputPass.js";
+
+AFRAME.registerComponent("bloom", {
+  schema: {
+    threshold: { type: "number", default: 1 },
+    strength: { type: "number", default: 0.5 },
+    radius: { type: "number", default: 1 },
+  },
+  events: {
+    rendererresize: function () {
+      this.renderer.getSize(this.size);
+      this.composer.setSize(this.size.width, this.size.height);
+    },
+  },
+  init() {
+    this.size = new THREE.Vector2();
+    this.scene = this.el.object3D;
+    this.renderer = this.el.renderer;
+    this.camera = this.el.camera;
+    this.bind();
+  },
+  update: function () {
+    if (this.composer) {
+      this.composer.dispose();
+    }
+    // create composer with multisampling to avoid aliasing
+    const resolution = this.renderer.getDrawingBufferSize(new THREE.Vector2());
+    const renderTarget = new THREE.WebGLRenderTarget(
+      resolution.width,
+      resolution.height,
+      { type: THREE.HalfFloatType, samples: 8 }
+    );
+
+    this.composer = new EffectComposer(this.renderer, renderTarget);
+
+    // create render pass
+    const renderScene = new RenderPass(this.scene, this.camera);
+    this.composer.addPass(renderScene);
+
+    // create bloom pass
+    const strength = this.data.strength;
+    const radius = this.data.radius;
+    const threshold = this.data.threshold;
+    if (this.bloomPass) {
+      this.bloomPass.dispose();
+    }
+    this.bloomPass = new UnrealBloomPass(
+      resolution,
+      strength,
+      radius,
+      threshold
+    );
+    this.composer.addPass(this.bloomPass);
+
+    // create output pass
+    if (this.outputPass) {
+      this.outputPass.dispose();
+    }
+    this.outputPass = new OutputPass();
+    this.composer.addPass(this.outputPass);
+  },
+
+  bind: function () {
+    this.originalRender = this.el.renderer.render;
+    const self = this;
+    let isInsideComposerRender = false;
+
+    this.el.renderer.render = function () {
+      if (isInsideComposerRender) {
+        self.originalRender.apply(this, arguments);
+      } else {
+        isInsideComposerRender = true;
+        self.composer.render(self.el.sceneEl.delta / 1000);
+        isInsideComposerRender = false;
+      }
+    };
+  },
+
+  remove() {
```

Comment:
should be `function: remove () {`

Reply:
Note that you can press this button in review to suggest changes as I did.

<img width="280" alt="{FE44035C-8CB4-45BC-B600-2DB36D504E27}" src="https://github.com/user-attachments/assets/c5e1d205-46c9-4be0-9e39-403a56338efb" />

Reply:
yeah. I can edit the PR directly too. Deliberate to make people do the changes. Especially first-time contributors :)

Reply:
omg I always wondered how folks made suggestions, I saw that only on a few PRs, I never saw that icon myself, no idea to look at the style toolbar for it.
I still have no idea how you commit directly to a PR before merge @dmarcos

Reply:
I think only owners of a repo can add commits to a PR but not 100% sure

Reply:
Yes obviously, but where and how?

Reply:
Oh like this https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally
I always do the complicated way, add the remote, fetch the remote and get the branch on the command line. I had no idea you could just get the PR branch lol.

Reply:
I'm reading the doc again, it's been years since I read those pages.
There is a screenshot with the suggest change icon:
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request

Reply:
I just edit the file on the PR directly on Github. living on the edge 😅